include/deal.II/multigrid/mg_transfer.h

```diff
@@ -48,7 +52,7 @@ namespace internal
     typedef ::dealii::SparseMatrix<typename VectorType::value_type> Matrix;
 
     template <typename SparsityPatternType, typename DoFHandlerType>
-    static void reinit(Matrix &matrix, Sparsity &sparsity, int level, const SparsityPatternType &sp, const DoFHandlerType &)
+    static void reinit(Matrix &matrix, Sparsity &sparsity, int level, SparsityPatternType &sp, const DoFHandlerType &)
```

Comment:
I must admit I'm not overly happy to break the interface just because PETSc is not able of distributing the sparsity pattern. Shouldn't we simply move the `distribute_sparsity_pattern` to the site where `reinit` is called? That would be much cleaner, even though we would probably need to get the MPI communicator there...

Reply:
I think the only call site is MGTransferPrebuilt<...>::build_matrices(...). That method has access to the DoFHandler, so the communicator could just be retrieved from the triangulation like it is done inside the MatrixSelector<...>::reinit method. This would also have the added advantage that then the reinit method for PETSc could accept arbitrary sparsity pattern types, just like the others. Currently it can only use DynamicSparsityPattern because the other types don't have a row index set accessor. If you give go ahead, I will rework this as suggested and revert the interface.

Reply:
Yes, this is exactly what I was thinking about!